internal/cmd/controller/agentmanagement/controllers/cluster/import.go

```diff
@@ -291,7 +305,10 @@ func (i *importHandler) importCluster(cluster *fleet.Cluster, status fleet.Clust
 
 	if apiServerURL == "" {
 		if len(cfg.APIServerURL) == 0 {
-			return status, fmt.Errorf("missing apiServerURL in fleet config for cluster auto registration")
+			// Current config cannot be deployed, so remove the "config changed" mark
+			logrus.Warnf("cannot import cluster '%s/%s', missing apiServerURL in fleet config for cluster auto registration", cluster.Namespace, cluster.Name)
+			status.AgentConfigChanged = false
```

Comment:
The problem with this is that the KubeconfigSecret could eventually be updated to contain a valid apiServerURL key, but the code that checks if changes are needed is `onConfig`, which won't be triggered.

Reply:
Solved by adding a new handler on Secrets that will retrieve all Clusters referencing a Secret (efficiently, using an indexer), then trigger the config check for them.